webob/response.py

```diff
@@ -1287,13 +1296,11 @@ def __call__(self, environ, start_response):
         if self.conditional_response:
             return self.conditional_response_app(environ, start_response)
 
-        # Build headerlist, if there is a Location header, make it absolute
-        headerlist = [
-            (k, v) if k.lower() != 'location'
-            else (k, self._make_location_absolute(environ, v))
-            for (k, v)
-            in self._headerlist
-        ]
+        # If the HTTP Status code might have a Location header, try to make it absolute
+        if self._status[:3] in ('201', '301', '302', '303', '305', '307'):
```

Comment:
For starters this doesn't support 308. In general, I don't like this change. It's actually fairly difficult to figure out what responses support Location... you're only checking right now for the responses that **require** Location. It's complicated.